### PR TITLE
Cleaned up crates merging & added 1 to publisher ind

### PR DESCRIPTION
**Pull request by Owez**
Cleaned up publisher HashMap merging which had a comment about it & fixed a small bug where the progress started at `(0/x)` instead of `(1/x)`.

### CODE REVIEW

File: src/subcommands/crates.rs [hunk 1]
```diff
@@ -1,20 +1,13 @@
...
 
-    // Merge maps back together. Ewww. Maybe there's a better way to go about this.
-    let mut owners: HashMap<String, Vec<PublisherData>> = HashMap::new();
-    for map in &[publisher_users, publisher_teams] {
-        for (crate_name, publishers) in map.iter() {
-            let entry = owners.entry(crate_name.clone()).or_default();
-            entry.extend_from_slice(publishers);
-        }
+    for team in publisher_teams {
+        owners.entry(team.0).or_default().extend(team.1)
```
Comment (Shnatsel) [inline src/subcommands/crates.rs:10]:
```suggestion
    for (crate_name, publishers) in publisher_teams {
        owners.entry(crate_name).or_default().extend(publishers)
```

**Review by Shnatsel — COMMENTED** [review]
Thanks! One readability issue (I've left a suggestion on how to fix it), otherwise looks good to me!

File: src/subcommands/crates.rs (hunk 1 above)
Comment (Owez) [reply src/subcommands/crates.rs]: Ah yeah, good spot

**Comment by Shnatsel** [comment]
Thanks!

**Pull request merged**